### PR TITLE
Remove legacy sintaxis.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

**Review by vjik — CHANGES_REQUESTED** [review]

**Review by vjik — CHANGES_REQUESTED** [review]
Need changelog

File: composer.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
         "spatie/phpunit-watcher": "^1.23",
         "vimeo/psalm": "^4.30|^5.6",
         "yiisoft/aliases": "^1.1|^2.0",
+        "yiisoft/cache": "^2.0",
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply composer.json:39]: in another pr.

**Review by vjik — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,8 @@
 
 ## 1.0.1 under development
 
-- no changes in this release.
+- Bug #722: Remove legacy sintaxis for array typecast in Param class. (@terabytesoftw)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- Chg #722: Remove legacy array syntax for typecast. Use `Param` instead (@terabytesoftw)
```

Need verify.

Comment (vjik) [reply CHANGELOG.md:5]: Is major change?

Comment (darkdef) [reply CHANGELOG.md]: I think - no. It's remove unused code

**Review by darkdef — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**